src/pyuvdata/beam_interface.py

```diff
@@ -82,6 +90,105 @@ def __post_init__(self, include_cross_pols: bool):
                     "efield beam, either provide an efield UVBeam or do not "
                     "specify `beam_type`."
                 )
+        elif self.beam_type is None:
+            self.beam_type = "efield"
+
+    @property
+    def Npols(self):
+        """The number of polarizations in the beam."""
+        return self.beam.Npols or len(self.polarization_array)
+
+    @property
+    def polarization_array(self):
+        """The polarizations defined on the beam."""
+        return self.beam.polarization_array
+
+    @property
+    def feed_array(self):
+        """The feeds for which the beam is defined."""
+        return self.beam.feed_array
+
+    @property
+    def Nfeeds(self):
+        """The number of feeds defined on the beam."""
+        return self.beam.Nfeeds or len(self.feed_array)
+
+    def clone(self, **kw):
+        """Return a new instance with updated parameters."""
+        return replace(self, **kw)
+
+    def as_power_beam(
+        self, include_cross_pols: bool = True, allow_beam_mutation: bool = False
+    ):
+        """Return a new interface instance that is in the power-beam mode.
+
+        If already in the power-beam mode, this is a no-op. Note that this might be
+        slighty unexpected, because the effect of `include_cross_pols` is not accounted
+        for in this case.
```

Comment:
I think we should issue a warning in that case. We can tell what was set before because if `include_cross_pols` was False when it was initialized, Npols is equal to Nfeeds. If it was True, Npols is > Nfeeds. I guess if there's only one feed, the value of this parameter doesn't matter and the warning shouldn't be raised.